src/components/common/withLocation.jsx

```diff
@@ -0,0 +1,21 @@
+import React from "react";
+import { Location } from "@reach/router";
+
+// Based on https://medium.com/@chrisfitkin/how-to-get-query-string-parameter-values-in-gatsby-f714161104f
+
+/* eslint-disable react/jsx-props-no-spreading - explicitly use property spreading for higher-order component (a component wrapping another component) */
+const withLocation = (ComponentToWrap) => (props) =>
```

Comment:
I think this works for any component due to using the router directly, whereas with Gatsby's built-in feature only pages (and templates) can receive the `location` parameter.